src/Compilers/CSharp/Test/Semantic/Semantics/BindingAsyncTests.cs

```diff
@@ -1222,6 +1222,31 @@ interface IInterface
         }
 
         [Fact]
+        public void BackCompatAsyncMain()
+        {
+            var source = @"
+using System.Threading.Tasks;
+
+class A
+{
+    async static Task Main(string[] args)
+    {
+        await Task.Factory.StartNew(() => { });
+    }
+    static void Main(){}
+}";
+
+            foreach (var langVersion in new LanguageVersion[] { LanguageVersion.CSharp7, LanguageVersion.CSharp7_1 })
+            {
+                var compilation = CreateCompilationWithMscorlib45(source, options: TestOptions.DebugExe, parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp7_1));
+                compilation.VerifyDiagnostics(
+                    // (6,23): error CS0017: Program has more than one entry point defined. Compile with /main to specify the type that contains the entry point.
```

Comment:
These tests are removed and have been added to the test plan.

---
In reply to: [111525364](https://github.com/dotnet/roslyn/pull/18472#discussion_r111525364) [](ancestors = 111525364)